### PR TITLE
wrapping store registration

### LINKED ISSUE #8823: [FR] Refresh MLFLOW_TRACKING_TOKEN
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

As of now, MLFlow does not support the use case of registering a long-building artefact with a bearer token.

To my best understanding, `MLFLOW_TRACKING_TOKEN` is set at the beginning of the client session. 

My proposal is to implement re-reading the token on demand.

Note: Extending token life is not an option – possible only for some use-cases and anyways limited.

### Motivation

> #### What is the use case for this feature?

As a user I want to upload long-building artefacts with bearer tokens.

> #### Why is this use case valuable to support for MLflow users in general?

A common challenge when prototyping on large ML models in the cloud.

> #### Why is this use case valuable to support for your project(s) or organization?

Enables scalability of MLFlow.

> #### Why is it currently difficult to achieve this use case?

Not supported / achievable by hacks.

### Details

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### LINKED ISSUE #8880: clean and transparent store registration
These lines

https://github.com/mlflow/mlflow/blob/152063e0b6fbadfbb2baecaf7d0ac7ca1b304b31/mlflow/tracking/_tracking_service/utils.py#L194C1-L206C1

initialize a global variable holding important information about store endpoints. 

But the style is not good:
- should be moved to the top and marked clearly
- perhaps it would be beneficial to wrap them to be recycled per user request (re-register / re-initialize)

### PR DESCRIPTION
**Pull request by maciejskorski**
# Description

## Summary

This PR **wraps and exposes the structure holding endpoint data**, stored in `mlflow.tracking._tracking_service.utils.py`, so that it can be 

1. initialized elegantly (improvement of the code quality) 
2. refreshed per user request (functionality useful when prototyping in cloud)

This example shows how to use this change to refresh tokens:
```python
import mlflow
from mlflow.tracking._tracking_service.utils import _register_tracking_stores
import os

os.environ['MLFLOW_TRACKING_TOKEN'] = my_first_token
# .. long and intensive mlflow usage, so that token expires

# refresh token
os.environ['MLFLOW_TRACKING_TOKEN'] = my_second_token  # e.g. gcloud auth print-identity-token
register_tracking_stores() # updates the token !

# keep working under refreshed token
mlflow.get_experiment_by_name('test_mlflow')
``` 

 
Impact:
-  Resolves #8823 
- Resolves #8880

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by maciejskorski** [comment]
> Please sign off your commit(s)

I think they have been signed, no?

File: mlflow/tracking/_tracking_service/utils.py [hunk 1]
```diff
@@ -190,19 +190,28 @@ def _get_databricks_uc_rest_store(store_uri, **_):
     )
 
 
+global _tracking_store_registry
 _tracking_store_registry = TrackingStoreRegistry()
```
Comment (WeichenXu123) [inline mlflow/tracking/_tracking_service/utils.py:194]: Do we need this change ? `_tracking_store_registry` is already a global variable.

File: mlflow/tracking/_tracking_service/utils.py [hunk 2]
```diff
@@ -190,19 +190,28 @@ def _get_databricks_uc_rest_store(store_uri, **_):
...
-_tracking_store_registry.register("file", _get_file_store)
-_tracking_store_registry.register("databricks", _get_databricks_rest_store)
-_tracking_store_registry.register(_DATABRICKS_UNITY_CATALOG_SCHEME, _get_databricks_uc_rest_store)
 
-for scheme in ["http", "https"]:
-    _tracking_store_registry.register(scheme, _get_rest_store)
 
-for scheme in DATABASE_ENGINES:
-    _tracking_store_registry.register(scheme, _get_sqlalchemy_store)
+def register_tracking_stores():
```
Comment (WeichenXu123) [inline mlflow/tracking/_tracking_service/utils.py:197]: can we mark it as internal method ? like `_register_tracking_stores`

File: mlflow/tracking/_tracking_service/utils.py (hunk 1 above)
Comment (maciejskorski) [reply mlflow/tracking/_tracking_service/utils.py]: Right. Changed.

File: mlflow/tracking/_tracking_service/utils.py (hunk 2 above)
Comment (maciejskorski) [reply mlflow/tracking/_tracking_service/utils.py]: Good point. Changed.

File: mlflow/tracking/_tracking_service/utils.py [hunk 3]
```diff
@@ -191,18 +191,27 @@ def _get_databricks_uc_rest_store(store_uri, **_):
...
-_tracking_store_registry.register(_DATABRICKS_UNITY_CATALOG_SCHEME, _get_databricks_uc_rest_store)
 
-for scheme in ["http", "https"]:
-    _tracking_store_registry.register(scheme, _get_rest_store)
 
-for scheme in DATABASE_ENGINES:
-    _tracking_store_registry.register(scheme, _get_sqlalchemy_store)
+def _register_tracking_stores():
+    """Register tracking stores.
+    This method is to be called upon the module initialization or per user's request to refresh"""
```
Comment (harupy) [inline mlflow/tracking/_tracking_service/utils.py:198]: Do we need this docstring?

Comment (harupy) [reply mlflow/tracking/_tracking_service/utils.py:198]: We normally don't expect users to call this function because it's in a private module.

Comment (maciejskorski) [reply mlflow/tracking/_tracking_service/utils.py:198]: My style is verbose. 

I am happy to remove if you think it is self-descriptive enough?

Comment (harupy) [reply mlflow/tracking/_tracking_service/utils.py:198]: Yeah, I think it's  self-descriptive enough.

Comment (maciejskorski) [reply mlflow/tracking/_tracking_service/utils.py]: changed in 4e4e47254cb86e6b4cba60b3a019e89f250b0ad7

**Review by harupy — APPROVED** [review]

**Comment by harupy** [comment]
LGTM!

**Comment by maciejskorski** [comment]
> LGTM!

We should perhaps wait for @WeichenXu123's opinion, it would be rude to self-resolve comments.

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Pull request merged**